### PR TITLE
Properties transformer

### PR DESCRIPTION
**Pull request by Cerfoglg**
Related to https://github.com/benchflow/collectors/issues/31
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-211061958%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-211856684%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-212448170%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-219061291%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-220177709%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-220269189%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-220286939%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-220287013%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63614114%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63614569%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63793331%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63793445%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63793589%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63793726%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63793880%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63844934%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63854922%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63855016%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23issuecomment-211061958%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40Cerfoglg%20map%20the%20related%20issues%2C%20please.%20We%20are%20gonna%20merge%20this%2C%20when%20it%20is%20integrated%20with%20tests%20and%20in%20the%20final%20state%20of%20refactoring.%22%2C%20%22created_at%22%3A%20%222016-04-17T17%3A16%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20Ah%2C%20the%20issue%20regarding%20properties%20was%20only%20opened%20for%20the%20collector%20of%20the%20properties.%20See%20the%20pull%20I%20made%20for%20the%20collectors%20regarding%20it%22%2C%20%22created_at%22%3A%20%222016-04-19T10%3A50%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22Moreover%20make%20sure%20that%20you%20save%20only%20once%20the%20same%20%60host_properties%60.%20There%20should%20be%20a%20single%20value%20that%20let%20you%20identify%20if%20the%20host%20is%20the%20same%2C%20e.g.%2C%20%60ID%60%22%2C%20%22created_at%22%3A%20%222016-04-20T14%3A29%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22-%20Move%20code%20away%20from%20the%20main%22%2C%20%22created_at%22%3A%20%222016-05-13T14%3A35%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40Cerfoglg%20ping%20me%20when%20it%27s%20ready%20to%20the%20final%20review%22%2C%20%22created_at%22%3A%20%222016-05-18T22%3A31%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20Should%20be%20ready%20now%22%2C%20%22created_at%22%3A%20%222016-05-19T09%3A09%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-05-19T10%3A30%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-05-19T10%3A30%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20363462c7879c685413dead534ec2e70bbc1644f4%20data-transformers/transformers/propertiesTransformer.py%20184%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63614114%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20guess%20this%20is%20a%20common%20function%20to%20multiple%20transformers.%20Are%20you%20going%20to%20refactor%20it%20in%20this%20pull%20request%20or%20in%20a%20new%20one%3F%20If%20it%27s%20a%20new%20one%2C%20open%20an%20issue%20about%20this%20and%20point%20to%20this%20comment%22%2C%20%22created_at%22%3A%20%222016-05-17T22%3A16%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-05-18T22%3A23%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20data-transformers/transformers/propertiesTransformer.py%3AL192-264%22%7D%2C%20%22Pull%20363462c7879c685413dead534ec2e70bbc1644f4%20data-transformers/transformers/propertiesTransformer.py%20195%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63793589%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20should%20go%20in%20a%20%60commons%60%20library%20of%20%60data-transformers%60.%20I%20marked%20it%20in%20%235%20%22%2C%20%22created_at%22%3A%20%222016-05-18T22%3A25%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-05-18T22%3A26%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20data-transformers/transformers/propertiesTransformer.py%3AL1-270%22%7D%2C%20%22Pull%20363462c7879c685413dead534ec2e70bbc1644f4%20data-transformers/transformers/propertiesTransformer.py%20244%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63793445%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Answer%20to%20the%20following%2C%20please%3A%20https%3A//github.com/benchflow/data-transformers/pull/50/files/4f9492676ef254ee77e9ad986fea2117cd1125e3..363462c7879c685413dead534ec2e70bbc1644f4%23r63614569%22%2C%20%22created_at%22%3A%20%222016-05-18T22%3A24%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-05-19T10%3A29%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20data-transformers/transformers/propertiesTransformer.py%3AL1-270%22%7D%2C%20%22Pull%20363462c7879c685413dead534ec2e70bbc1644f4%20data-transformers/transformers/propertiesTransformer.py%20191%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/50%23discussion_r63614569%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20do%20you%20need%20to%20access%20to%20the%20first%20object%20in%20the%20returned%20array%3F%20What%20is%20the%20returned%20array%3F%22%2C%20%22created_at%22%3A%20%222016-05-17T22%3A20%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-05-18T22%3A27%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20It%27s%20a%20list%20containing%20all%20the%20lines%20in%20the%20file.%20But%20these%20files%20only%20have%20one%20line%2C%20so%20I%20just%20take%20that%22%2C%20%22created_at%22%3A%20%222016-05-19T09%3A09%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-05-19T10%3A28%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20data-transformers/transformers/propertiesTransformer.py%3AL192-264%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/VincenzoFerme%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/VincenzoFerme'></a>
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/50#issuecomment-211061958'>General Comment</a></b>
- <a href='https://github.com/VincenzoFerme'></a> @Cerfoglg map the related issues, please. We are gonna merge this, when it is integrated with tests and in the final state of refactoring.
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme Ah, the issue regarding properties was only opened for the collector of the properties. See the pull I made for the collectors regarding it
- <a href='https://github.com/VincenzoFerme'></a> Moreover make sure that you save only once the same `host_properties`. There should be a single value that let you identify if the host is the same, e.g., `ID`
- <a href='https://github.com/Cerfoglg'></a> - Move code away from the main
- <a href='https://github.com/VincenzoFerme'></a> @Cerfoglg ping me when it's ready to the final review
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme Should be ready now
- [x] <a href='#crh-comment-Pull 363462c7879c685413dead534ec2e70bbc1644f4 data-transformers/transformers/propertiesTransformer.py 184'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/50#discussion_r63614114'>File: data-transformers/transformers/propertiesTransformer.py:L192-264</a></b>
- <a href='https://github.com/VincenzoFerme'></a> I guess this is a common function to multiple transformers. Are you going to refactor it in this pull request or in a new one? If it's a new one, open an issue about this and point to this comment
- [x] <a href='#crh-comment-Pull 363462c7879c685413dead534ec2e70bbc1644f4 data-transformers/transformers/propertiesTransformer.py 191'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/50#discussion_r63614569'>File: data-transformers/transformers/propertiesTransformer.py:L192-264</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Why do you need to access to the first object in the returned array? What is the returned array?
- <a href='https://github.com/VincenzoFerme'></a> :+1:
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme It's a list containing all the lines in the file. But these files only have one line, so I just take that
- [x] <a href='#crh-comment-Pull 363462c7879c685413dead534ec2e70bbc1644f4 data-transformers/transformers/propertiesTransformer.py 244'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/50#discussion_r63793445'>File: data-transformers/transformers/propertiesTransformer.py:L1-270</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Answer to the following, please: https://github.com/benchflow/data-transformers/pull/50/files/4f9492676ef254ee77e9ad986fea2117cd1125e3..363462c7879c685413dead534ec2e70bbc1644f4#r63614569
- [x] <a href='#crh-comment-Pull 363462c7879c685413dead534ec2e70bbc1644f4 data-transformers/transformers/propertiesTransformer.py 195'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/50#discussion_r63793589'>File: data-transformers/transformers/propertiesTransformer.py:L1-270</a></b>
- <a href='https://github.com/VincenzoFerme'></a> This should go in a `commons` library of `data-transformers`. I marked it in #5

<a href='https://www.codereviewhub.com/benchflow/data-transformers/pull/50?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/benchflow/data-transformers/pull/50?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/benchflow/data-transformers/pull/50?approve=0'></a>&nbsp;<a href='https://github.com/benchflow/data-transformers/pull/50'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by VincenzoFerme** [comment]
@Cerfoglg map the related issues, please. We are gonna merge this, when it is integrated with tests and in the final state of refactoring.

**Comment by Cerfoglg** [comment]
@VincenzoFerme Ah, the issue regarding properties was only opened for the collector of the properties. See the pull I made for the collectors regarding it

**Comment by VincenzoFerme** [comment]
Moreover make sure that you save only once the same `host_properties`. There should be a single value that let you identify if the host is the same, e.g., `ID`

**Comment by Cerfoglg** [comment]
- Move code away from the main

File: data-transformers/transformers/propertiesTransformer.py [hunk 1]
```diff
@@ -189,74 +192,72 @@ def createVersionDict(a):
...
-        .setMaster(sparkMaster) \
-        .set("spark.cassandra.connection.host", cassandraHost)
+    conf = SparkConf().setAppName("Properties Transformer")
     sc = CassandraSparkContext(conf=conf)
     
-    res = urllib2.urlopen("http://"+minioHost+":"+minioPort+"/"+filePath+"_inspect.gz")
-    compressed = io.BytesIO(res.read())
-    decompressed = gzip.GzipFile(fileobj=compressed)
-    inspectData = decompressed.readline()
+    # Retrieve the configuration file that was sent to spark
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:237]: I guess this is a common function to multiple transformers. Are you going to refactor it in this pull request or in a new one? If it's a new one, open an issue about this and point to this comment

File: data-transformers/transformers/propertiesTransformer.py [hunk 2]
```diff
@@ -189,74 +192,72 @@ def createVersionDict(a):
...
+    confPath = SparkFiles.get("data-transformers.yml")
+    with open(confPath) as f:
+        transformerConfiguration = yaml.load(f)
+        cassandraKeyspace = transformerConfiguration["cassandra_keyspace"]
+        minioPort = transformerConfiguration["minio_port"]
     
+    inspectData = getFromMinio("http://"+minioHost+":"+minioPort+"/"+filePath+"_inspect.gz")[0]
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:244]: Why do you need to access to the first object in the returned array? What is the returned array?

File: data-transformers/transformers/propertiesTransformer.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        if data[0] == "Os":
+            d["docker_os"] = data[1]
+            continue
+        if data[0] == "ApiVersion":
+            d["docker_api_version"] = data[1]
+            continue
+        if data[0] == "GitCommit":
+            d["docker_git_commit"] = data[1]
+            continue
+        if data[0] == "GoVersion":
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:184]: :+1:

File: data-transformers/transformers/propertiesTransformer.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    sc = CassandraSparkContext(conf=conf)
+    
+    # Retrieve the configuration file that was sent to spark
+    confPath = SparkFiles.get("data-transformers.yml")
+    with open(confPath) as f:
+        transformerConfiguration = yaml.load(f)
+        cassandraKeyspace = transformerConfiguration["cassandra_keyspace"]
+        minioPort = transformerConfiguration["minio_port"]
+    
+    inspectData = getFromMinio("http://"+minioHost+":"+minioPort+"/"+filePath+"_inspect.gz")[0]
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:244]: Answer to the following, please: https://github.com/benchflow/data-transformers/pull/50/files/4f9492676ef254ee77e9ad986fea2117cd1125e3..363462c7879c685413dead534ec2e70bbc1644f4#r63614569

File: data-transformers/transformers/propertiesTransformer.py [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+            continue
+        if data[0] == "Arch":
+            d["docker_arch"] = data[1]
+            continue
+        if data[0] == "KernelVersion":
+            d["docker_kernel_version"] = data[1]
+            continue
+    return d
+
+def getFromMinio(url):
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:195]: This should go in a `commons` library of `data-transformers`. I marked it in #5

Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:195]: :+1:

File: data-transformers/transformers/propertiesTransformer.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+            d["docker_git_commit"] = data[1]
+            continue
+        if data[0] == "GoVersion":
+            d["docker_go_version"] = data[1]
+            continue
+        if data[0] == "Arch":
+            d["docker_arch"] = data[1]
+            continue
+        if data[0] == "KernelVersion":
+            d["docker_kernel_version"] = data[1]
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:191]: :+1:

**Comment by VincenzoFerme** [comment]
@Cerfoglg ping me when it's ready to the final review

File: data-transformers/transformers/propertiesTransformer.py (hunk 2 above)
Comment (Cerfoglg) [inline data-transformers/transformers/propertiesTransformer.py:244]: @VincenzoFerme It's a list containing all the lines in the file. But these files only have one line, so I just take that

**Comment by Cerfoglg** [comment]
@VincenzoFerme Should be ready now

File: data-transformers/transformers/propertiesTransformer.py (hunk 6 above)
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:191]: :+1:

File: data-transformers/transformers/propertiesTransformer.py (hunk 4 above)
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py:244]: :+1:

**Comment by VincenzoFerme** [comment]
:+1:

**Comment by VincenzoFerme** [comment]
:shipit:

**Pull request merged**